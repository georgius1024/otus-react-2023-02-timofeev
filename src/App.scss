.layout {
  display: flex;
  height: 100vh;
  width: 100vw;
  overflow: hidden;
  flex-direction: column;
}
.main {
  flex-grow: 1;
  display: flex;
  align-items: center;
  justify-content: center;
}
.header {
  background-color: darkblue;
  color: #fff;
  display: flex;
  flex-wrap: wrap;
  padding: 0;
  a {
    color: #fff;
  }
  .section {
    min-width: 240px;
    padding: 32px;
    display: flex;
  }
  .section-1, .section-2, .section-3 {
    width: 33.33%;
  }
  .section-1 {
    justify-content: flex-start;
  }
  .section-2 {
    justify-content: center;
  }
  .section-3 {
    justify-content: flex-end;
  }
}

.footer {
  background-color: darkblue;
  color: #fff;
  display: flex;
  flex-wrap: wrap;
  padding: 16px 32px;
}
.wrappable {
  max-width: 360px;
  text-overflow: ellipsis;
  overflow-x: hidden;
}

.rejecting {
  animation: 1s ease-in-out rejecting;
}

@keyframes rejecting {
  0% {
    transform: translateX(0);
  }
  6.5% {
    transform: translateX(-6px) rotateY(-9deg);
  }
  18.5% {
    transform: translateX(5px) rotateY(7deg);
  }
  31.5% {
    transform: translateX(-3px) rotateY(-5deg);
  }
  43.5% {
    transform: translateX(2px) rotateY(3deg);
  }
  50% {
    transform: translateX(0);
  }
}

