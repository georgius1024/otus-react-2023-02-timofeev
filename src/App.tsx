import { useState, ReactElement } from "react";
import { useSelector } from "react-redux";
import {
  createBrowserRouter,
  RouterProvider,
  Navigate,
} from "react-router-dom";

import "@/App.scss";
import Layout from "@/components/Layout";
import Login from "@/pages/Login";
import Forgot from "@/pages/Forgot";
import Register from "@/pages/Register";
import Todos from "@/pages/Todos";
import TodoPage from "@/pages/TodoPage";

const router = createBrowserRouter([
  {
    path: "/",
    element: (
      <Layout>
        <Todos />
      </Layout>
    ),
  },
  {
    path: "/:id",
    element: (
      <Layout>
        <TodoPage />
      </Layout>
    ),
  },
  {
    path: "/login",
    element: (
      <Layout>
        <Login />
      </Layout>
    ),
  },
  {
    path: "/register",
    element: (
      <Layout>
        <Register />
      </Layout>
    ),
  },
  {
    path: "/forgot",
    element: (
      <Layout>
        <Forgot />
      </Layout>
    ),
  },
]);

function App(): ReactElement {
  return <RouterProvider router={router} />;
}

export default App;
