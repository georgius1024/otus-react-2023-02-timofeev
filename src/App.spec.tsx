import { render, screen } from "@testing-library/react";

import App from "@/App";

describe("App", () => {
  it("renders headline", () => {
    expect(render(<App />)).toMatchSnapshot();
  });
});
