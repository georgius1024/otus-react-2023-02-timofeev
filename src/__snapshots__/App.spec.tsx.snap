// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > renders headline 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="layout"
      >
        <header
          class="header"
        >
          <div
            class="section section-1"
          >
            <a
              href="/"
            >
              [LOGO]
            </a>
          </div>
          <div
            class="section section-2"
          >
            TODOAPP
          </div>
          <div
            class="section section-3"
          >
            <a
              href="/login"
            >
              Login
            </a>
             
            <a
              href="/register"
            >
              Register
            </a>
          </div>
        </header>
        <main
          class="main"
        >
          <form
            name="login"
          >
            <h1>
              Login
            </h1>
            <label>
              Email
              <input
                name="email"
                type="email"
                value=""
              />
            </label>
            <label>
              Password
              <input
                type="password"
                value=""
              />
            </label>
            <a
              href="/forgot"
            >
              Forgot password?
            </a>
            <button
              type="button"
            >
              Login
            </button>
          </form>
        </main>
        <footer
          class="footer"
        >
          (c) Diligent student
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="layout"
    >
      <header
        class="header"
      >
        <div
          class="section section-1"
        >
          <a
            href="/"
          >
            [LOGO]
          </a>
        </div>
        <div
          class="section section-2"
        >
          TODOAPP
        </div>
        <div
          class="section section-3"
        >
          <a
            href="/login"
          >
            Login
          </a>
           
          <a
            href="/register"
          >
            Register
          </a>
        </div>
      </header>
      <main
        class="main"
      >
        <form
          name="login"
        >
          <h1>
            Login
          </h1>
          <label>
            Email
            <input
              name="email"
              type="email"
              value=""
            />
          </label>
          <label>
            Password
            <input
              type="password"
              value=""
            />
          </label>
          <a
            href="/forgot"
          >
            Forgot password?
          </a>
          <button
            type="button"
          >
            Login
          </button>
        </form>
      </main>
      <footer
        class="footer"
      >
        (c) Diligent student
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
